application/src/main/java/org/togetherjava/tjbot/features/javamail/RSSHandlerRoutine.java

```diff
@@ -0,0 +1,412 @@
+        this.interval = this.config.rssPollInterval();
+        this.database = database;
+        this.fallbackChannelPattern =
+                Pattern.compile(this.config.fallbackChannelPattern()).asMatchPredicate();
+        this.targetChannelPatterns = new HashMap<>();
+        this.config.feeds().forEach(feed -> {
+            if (feed.targetChannelPattern() != null) {
+                Predicate<String> predicate =
+                        Pattern.compile(feed.targetChannelPattern()).asMatchPredicate();
+                targetChannelPatterns.put(feed, predicate);
+            }
+        });
+        this.rssReader = new RssReader();
+    }
+
+    @Override
+    @NotNull
+    public Schedule createSchedule() {
+        return new Schedule(ScheduleMode.FIXED_DELAY, 0, interval, TimeUnit.MINUTES);
+    }
+
+    @Override
+    public void runRoutine(@Nonnull JDA jda) {
+        this.config.feeds().forEach(feed -> sendRSS(jda, feed));
+    }
+
+    /**
+     * Sends all the necessary posts from a given RSS feed.
+     * <p>
+     * This handles fetching the latest posts from the given URL, checking which ones have already
+     * been posted by reading information from the database and updating the last posted date.
+     *
+     * @param jda The JDA instance.
+     * @param feedConfig The configuration object for the RSS feed.
+     */
+    private void sendRSS(JDA jda, RSSFeed feedConfig) {
+        TextChannel textChannel = getTextChannelFromFeed(jda, feedConfig).orElse(null);
+        if (textChannel == null) {
+            logger.warn("Tried to sendRss, got empty response (channel {} not found)",
+                    feedConfig.targetChannelPattern());
+            return;
+        }
+
+        List<Item> rssItems = fetchRSSItemsFromURL(feedConfig.url());
+        if (rssItems.isEmpty()) {
+            return;
+        }
+
+        // We only get the first RSS item from the list since we are
+        // assuming that the whole RSS feed uses the same format
+        if (!isValidDateFormat(rssItems.getFirst(), feedConfig)) {
+            logger.warn("Could not find valid date format for RSS feed {}", feedConfig.url());
+            return;
+        }
+
+        RssFeedRecord rssFeedRecord = getRssFeedRecordFromDatabase(feedConfig);
+        Optional<ZonedDateTime> lastSavedDate = getLastSavedDateFromDatabaseRecord(rssFeedRecord,
+                feedConfig.dateFormatterPattern());
+
+        ZonedDateTime lastPostedDate =
+                getLatestPostDateFromItems(rssItems, feedConfig.dateFormatterPattern());
+        updateLastDateToDatabase(feedConfig, rssFeedRecord, lastPostedDate);
+
+        if (lastSavedDate.isEmpty()) {
+            return;
+        }
+
+        final Predicate<Item> shouldItemBePosted = item -> {
+            ZonedDateTime itemPubDate =
+                    getDateTimeFromItem(item, feedConfig.dateFormatterPattern());
+            return itemPubDate.isAfter(lastSavedDate.get());
+        };
```

Comment:
This could be moved in its own method prepareItemPostPredicate or something like that.

Reply:
Are you talking about the entire highlight or certain parts of it?

Reply:
Entire highlight.